### ISSUE TITLE
Good job!

### ISSUE THREAD

**Issue by zh583007354** [JiahuiYu/slimmable_networks#7]
Exciting! I have studied your papers seriously. I am very interested in your work  and looking forward to your new code for the paper《AutoSlim...》.

Hope you can release it soon, please.

**Comment by JiahuiYu** [comment]
Hi @zh583007354 ,

Thanks for your interest. We are working on patent submissions and hopefully the full code will be released after that soon. Please stay tuned!

**Comment by zh583007354** [comment]
Thank you! But I want to konw where is the 'us_slimmable_training' mode in 'train.py' in current version? I can't find the codes about training mode as Algorithm 1 in "Universally Slimmable Networks and Improved Training Techniques".  There is only one condition 'slimmable_sample_training' to set the 'other_widths', but no training step as this, right?

**Comment by JiahuiYu** [comment]
@zh583007354 The training code of `Universally Slimmable Networks and Improved Training Techniques` is not yet released. Sorry for the confusion. The `train.py` almost remains the same for `Slimmable Neural Networks` and it works well if you only want to test the US-Nets.

**Comment by zh583007354** [comment]
Oh, so those codes will also be released until your patent submissions?  Should I need to write this training code myself?  In fact, I want to train my own model on a dataset about face recognition as my  Course Project.

**Comment by JiahuiYu** [comment]
Sounds good! I would encourage you to implement slimmable networks for face recognition. If you have any question during implementation, post your question here and I will try to help.

**Comment by zh583007354** [comment]
Thank you! I want to konw why you set "reduction" as "none" in "criterion = torch.nn.CrossEntropyLoss(reduction='none').cuda()" in train.py line 330. Another question is what is the difference between "Accumulate gradients" in Algorithm 1(US-Net) and "Compute gradients" in Algorithm 1(S-Net).

**Comment by JiahuiYu** [comment]
We set reduction=none because we will average the loss later. Both accumulate gradients and compute gradients is `loss.backward()`.

**Comment by wuzhiyang2016** [comment]
hello, do you have the plan of release the training code of Universally Slimmable Networks and Improved Training Techniques ?

**Comment by wuzhiyang2016** [comment]
very thanks, I'll have a try~
